pkg/dynamicinstrumentation/diconfig/config_manager.go

```diff
@@ -243,14 +243,17 @@ func (cm *RCConfigManager) readConfigs(r *ringbuf.Reader, procInfo *ditypes.Proc
 
 func applyConfigUpdate(procInfo *ditypes.ProcessInfo, probe *ditypes.Probe) {
 	log.Tracef("Applying config update: %v\n", probe)
-	err := AnalyzeBinary(procInfo)
-	if err != nil {
-		log.Errorf("couldn't inspect binary: %v\n", err)
-		return
+
+	if procInfo.TypeMap == nil {
+		err := AnalyzeBinary(procInfo)
+		if err != nil {
+			log.Errorf("couldn't inspect binary: %v\n", err)
+			return
+		}
```

Comment:
There's other fields which are set in procInfo from AnalyzeBinary besides the type map, so i'm not sure if this is the correct way to do this check. 

I originally thought that it would be an issue because when the binary is inspected we're only analyzing types for target functions, so if a config is updated, we wouldn't be able to parse types for the new configuration. However testing this branch doesn't show any issues.

What were the reservations you had about this particular change?

Reply:
Ok I did further investigation and realized this code doesn't belong here at all. `AnalyzeBinary` needs to be called for the sake of the RC config manager, but the memory config manager (i.e. what file based configuration is built on) has separate callbacks for when procs/configs are updated which call AnalyzeBinary already.

So remove this call to AnalyzeBinary all together and add it before the call to `applyConfigUpdate()` on line 239 (without the condition nil check)

Reply:
Done in c521706bedf6987961f29d07701111a8b3cac9ec.